File: LAB/P1/integrals-js/Integral.js

class Integral {
  constructor(init, end, step){
    this.init = init
    this.end = end
    this.step = step
  }

  integrate (expression) {
    let integral = 0
    for (let i = this.init; i <= this.end; i+=this.step){
      integral += expression.evaluate(i)*this.step
    }
    return integral
  }

}

module.exports = Integral

File: LAB/P1/integrals-java/Integral.java

class Integral {
  private double init;
  private double end;
  private double step;
  public Integral (double init, double end, double step){
    this.init = init;
    this.end = end;
    this.step = step;
  }

  public double integrate (Expresion expression) {
    double value = 0;
    for (double i = this.init; i < this.end; i+=this.step){
      value += expression.evaluate(i) * this.step;
    }
    return value;
  }
}